builtin/logical/ssh/path_roles.go

```diff
@@ -496,7 +499,7 @@ func (b *backend) pathRoleWrite(ctx context.Context, req *logical.Request, d *fr
 			Version:         roleEntryVersion,
 		}
 	} else if keyType == KeyTypeCA {
-		algorithmSigner := ""
+		algorithmSigner := DefaultAlgorithmSigner
```

Comment:
I might be mis-reading this code but at line 508 the case "": statement I would have assumed that we would have reset algorithmSigner back to DefaultAlgorithmSigner? Because we aren't I *think* we are writing out an empty value within the role's AlgorithmSigner with a V2 version

Reply:
Ah, you're right that we to allow users to make this explicitly default. 

I'm not sure I follow though -- we set algorithmSigner to whatever was passed in the request _if something was passed_... Which might be `default` or it might be the empty string. Technically there's a subtle behavior difference between the empty string (library default) and the literal `default` (our Vault default -- preferring SHA2-256 in the RSA case). So it is valid to pass an empty string via the API IMO.

Reply:
I see, explicit is better than implicit. Let's change it here to explicitly be default and have consistent behavior between the two.